seonoh/ArchitectureStudy/app/src/main/java/com/example/seonoh/seonohapp/CoinFragmentPresenter.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.seonoh.seonohapp
+
+import android.util.Log
+import com.example.seonoh.seonohapp.contract.CoinFragmentContract
+import com.example.seonoh.seonohapp.model.CurrentPriceInfoModel
+import com.example.seonoh.seonohapp.model.UseCoinModel
+import com.example.seonoh.seonohapp.repository.CoinRepositoryImpl
+import com.example.seonoh.seonohapp.util.CalculateUtils
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class CoinFragmentPresenter(
+    private val view: CoinFragmentContract.View
+) : CoinFragmentContract.Presenter {
+
+    private val coinRepository = CoinRepositoryImpl()
+    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
+
+    override fun loadData(marketName: String) {
+        compositeDisposable.addAll(
+            coinRepository.sendCurrentPriceInfo(marketName)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                    translateData(it)
+
+                }, { e ->
+                    Log.e("currentPriceInfo", "Network failed!! ${e.message}")
+                })
+        )
+    }
+
+    override fun clearDisposable() {
+        compositeDisposable.clear()
+    }
+
+    override fun setData(data: ArrayList<UseCoinModel>) {
+        view.initRecyclerView(data)
+    }
+
+    override fun translateData(result: List<CurrentPriceInfoModel>) {
+        var marketType = ""
```

Comment:
가능한 한 immutable 타입의 변수를 사용하는 것이 좋습니다.
```suggestion
        val marketType = = if (result.isNotEmpty()) {
            result[0].market.substringBefore("-")
        } else {
            ""
        }
```

Reply:
네 수정했습니다 !
e4a9883e0d0338b6ad6019092c809f2fd6a4f71b